datafusion/optimizer/src/common_subexpr_eliminate.rs

```diff
@@ -696,7 +696,8 @@ impl<'n> TreeNodeVisitor<'n> for ExprIdentifierVisitor<'_> {
             return Ok(TreeNodeRecursion::Jump);
         }
 
-        self.id_array.push((0, "".to_string()));
+        self.id_array
```

Comment:
Why do we need to store this boolean flag in `id_array` during the first traversal here? Can we just call `self.aliased |= matches!(expr, Expr::Alias(_))` during the second traversal in `f_down()` of `CommonSubexprRewriter`?

Reply:
To be completely honest, having it in the first traversal made the most sense to me, you may have noticed that the initial commit even used the `VisitStack`. But I like your suggestion better.